arcade/types/rect.py

```diff
@@ -0,0 +1,511 @@
+"""Rects all act the same, but take four of the possible eight attributes and calculate the rest."""
+from __future__ import annotations
+from typing import NamedTuple, Optional, TypedDict, Tuple
+
+from pyglet.math import Vec2
+
+from arcade.types.numbers import AsFloat
+from arcade.types.vector_like import AnchorPoint
+
+from arcade.utils import ReplacementWarning, warning
+
+RectParams = Tuple[AsFloat, AsFloat, AsFloat, AsFloat]
+ViewportParams = Tuple[int, int, int, int]
+
+
+class RectKwargs(TypedDict):
+    left: float
+    right: float
+    bottom: float
+    top: float
+    width: float
+    height: float
+    x: float
+    y: float
+
+
+class Rect(NamedTuple):
+    """Rects define a rectangle, with several convenience properties and functions.
+
+    This object is immutable by design. It provides no setters, and is a NamedTuple subclass.
+
+    Attempts to implement all Rectangle functions used in the library, and to be a helpful
+    tool for developers storing/maniulating rectangle and rectangle-like constructs.
+
+    Rectangles cannot rotate by design, since this complicates their implmentation a lot.
+
+    You probably don't want to create one of these directly, and should instead use a helper method, like
+    :py:func:`~arcade.types.rect.LBWH`, :py:func:`~arcade.types.rect.LRBT`, :py:func:`~arcade.types.rect.XYWH`,
+    :py:func:`~arcade.types.rect.AnchoredXYWH`, or :py:func:`~arcade.types.rect.Viewport`.
```

Comment:
:py:func:`~arcade.types.rect.AnchoredXYWH` gone

Reply:
Resolved in latest commit.